code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/contentcards/ContentCardSchemaDataUtils.kt

```diff
@@ -0,0 +1,138 @@
+     *
+     * @param map The map containing the data for [AepDismissButton].
+     * @return An instance of [AepDismissButton].
+     */
+    fun createAepDismissButton(map: Map<String, Any>): AepDismissButton {
+        return AepDismissButton(
+            style = map[MessagingConstants.AepUIKeys.STYLE] as? String ?: "NONE_ICON"
+        )
+    }
+}
+
+/**
+ * Parses the given [ContentCardSchemaData] to create an [AepUITemplate].
+ *
+ * @param contentCardSchemaData The content card schema data containing the content map.
+ * @return The parsed [AepUITemplate] or null if parsing fails.
+ * @throws IllegalArgumentException If the content map is null or if there is an error during parsing.
+ */
+fun getTemplateModelFromContentCardSchemaData(contentCardSchemaData: ContentCardSchemaData): AepUITemplate? {
+    val contentMap =
+        contentCardSchemaData.content as? Map<String, Any> ?: throw IllegalArgumentException("Content map is null")
+    try {
+        val id = contentMap[MessagingConstants.AepUIKeys.ID] as String
+        val title =
+            ContentCardSchemaDataUtils.createAepText(contentMap[MessagingConstants.AepUIKeys.TITLE] as Map<String, Any>)
+        val body = (contentMap[MessagingConstants.AepUIKeys.BODY] as? Map<String, Any>)?.let {
+            ContentCardSchemaDataUtils.createAepText(it)
+        }
+        val image = (contentMap[MessagingConstants.AepUIKeys.IMAGE] as? Map<String, Any>)?.let {
+            ContentCardSchemaDataUtils.createAepImage(it)
+        }
+        val actionUrl = contentMap[MessagingConstants.AepUIKeys.ACTION_URL] as? String
+        val buttons = (contentMap[MessagingConstants.AepUIKeys.BUTTONS] as? List<Map<String, Any>>)?.map {
+            ContentCardSchemaDataUtils.createAepButton(it)
+        }
+        val dismissBtn = (contentMap[MessagingConstants.AepUIKeys.DISMISS_BTN] as? Map<String, Any>)?.let {
+            ContentCardSchemaDataUtils.createAepDismissButton(it)
+        }
+        return SmallImageTemplate(
+            id = id,
+            title = title,
+            body = body,
+            image = image,
+            actionUrl = actionUrl,
+            buttons = buttons,
+            dismissBtn = dismissBtn
+        )
+    } catch (e: Exception) {
+        throw IllegalArgumentException("Error parsing content map: ${e.message}")
+    }
+}
+
+/**
+ * Provides an [AepUI] instance from the given [AepUITemplate].
+ *
+ * @param uiTemplate The template to convert into a UI component.
+ * @return An instance of [AepUI] representing the given template.
+ * @throws IllegalArgumentException If the provided template type is unsupported.
+ */
+fun provideAepUIfromAepUITemplate(uiTemplate: AepUITemplate): AepUI<*, *> {
+    return when (uiTemplate) {
+        is SmallImageTemplate -> {
+            SmallImageUI(uiTemplate, SmallImageCardUIState())
+        }
+        else -> throw IllegalArgumentException("Unsupported template type")
```

Comment:
Same comment here. Where is the exception caught? Can we handle this gracefully?

Reply:
Removed this function from scope of this PR

Reply:
Removed from the scope of this PR

Reply:
Return null when unsupported type